analytics/pubstack/pubstack_module_test.go

```diff
@@ -57,130 +55,165 @@ func loadJsonFromFile() (*analytics.AuctionObject, error) {
+			feature:     setUID,
+			logObject: func(module analytics.PBSAnalyticsModule) {
+				module.LogSetUIDObject(&analytics.SetUIDObject{Status: http.StatusOK})
+			},
+		},
+	}
 
+	for _, tt := range tests {
+		// original config is loaded when the module is created
+		// the feature is disabled so no events should be sent
+		origConfig := &Configuration{
+			Features: map[string]bool{
+				tt.feature: false,
+			},
+		}
+		// updated config is hot-reloaded after some time passes
+		// the feature is enabled so events should be sent
+		updatedConfig := &Configuration{
+			Features: map[string]bool{
+				tt.feature: true,
+			},
+		}
+
+		// create server with root endpoint that returns the current config
+		// add an intake endpoint that PBS hits when events are sent
+		rootCount := 0
+		mux := http.NewServeMux()
+		mux.HandleFunc("/", func(res http.ResponseWriter, req *http.Request) {
+			rootCount++
+			defer req.Body.Close()
+
+			if rootCount > 1 {
+				if data, err := json.Marshal(updatedConfig); err != nil {
+					res.WriteHeader(http.StatusBadRequest)
+				} else {
+					res.Write(data)
+				}
+			} else {
+				if data, err := json.Marshal(origConfig); err != nil {
+					res.WriteHeader(http.StatusBadRequest)
+				} else {
+					res.Write(data)
+				}
+			}
+		})
+
+		intakeChannel := make(chan int) // using a channel rather than examining the count directly to avoid race
+		mux.HandleFunc("/intake/"+tt.feature+"/", func(res http.ResponseWriter, req *http.Request) {
+			intakeChannel <- 1
+		})
+		server := httptest.NewServer(mux)
+		client := server.Client()
+
+		// set the server url on each of the configs
+		origConfig.Endpoint = server.URL
+		updatedConfig.Endpoint = server.URL
+
+		// instantiate module with 25ms config refresh rate
+		module, err := NewPubstackModule(client, "scope", server.URL, "25ms", 100, "1B", "10ms")
+		assert.Nil(t, err, tt.description)
+
+		// allow time for the module to load the original config
+		time.Sleep(10 * time.Millisecond)
+
+		pubstack, _ := module.(*PubstackModule)
+		// attempt to log but no event channel was created because the feature is disabled in the original config
+		tt.logObject(pubstack)
+
+		eventCount := 0
+
+		select {
+		case eventCount = <-intakeChannel:
+			assert.Fail(t, "an unexpected event was received", tt.description)
+		case <-time.After(10 * time.Millisecond):
+		}
+
+		assert.Equal(t, 0, eventCount, tt.description)
+
+		// allow time for the server to start serving the updated config
+		time.Sleep(10 * time.Millisecond)
+
+		// attempt to log; the event channel should have been created because the feature is enabled in updated config
+		tt.logObject(pubstack)
+
+		select {
+		case eventCount = <-intakeChannel:
+		case <-time.After(10 * time.Millisecond):
+			assert.Fail(t, "an event was expected but was never received", tt.description)
+		}
+
+		assert.Equal(t, 1, eventCount, tt.description)
```

Comment:
No longer needed. If you want to still have an assert to make the test look 'proper', you can create a helper method for asserting the channel, ie:

```
assertChanNone(intakeChannel) / assertChanOne(intakeChannel)
```

```
func assertChanNone(t TestingT, c <-chan int, msgAndArgs ...interface{}) bool {
  select {
    case <- c:
      return testify.Fail(t, "Should NOT receive an event, but did", msgAndArgs...)
    case <-time.After(10 * time.Millisecond):
      return true
  }
}

func assertChanOne(t TestingT, c <-chan int, msgAndArgs ...interface{}) bool {
  select {
    case <- c:
      return true
    case <-time.After(10 * time.Millisecond):
      return testify.Fail(t, "Should receive an event, but did NOT", msgAndArgs...)
  }
}
```